src/custom/utils/operator/error.ts

```diff
@@ -0,0 +1,78 @@
+// Conforms to backend API
+// https://github.com/gnosis/gp-v2-services/blob/0bd5f7743bebaa5acd3be13e35ede2326a096f14/orderbook/openapi.yml#L562
+export enum ApiErrorCodes {
+  DuplicateOrder = 'DuplicateOrder',
+  InvalidSignature = 'InvalidSignature',
+  MissingOrderData = 'MissingOrderData',
+  InsufficientValidTo = 'InsufficientValidTo',
+  InsufficientFunds = 'InsufficientFunds',
+  InsufficientFee = 'InsufficientFee',
+  UnsupportedToken = 'UnsupportedToken',
+  WrongOwner = 'WrongOwner'
+}
+
+export interface ApiError {
+  errorType: ApiErrorCodes
+  description: string
+}
+
+const API_ERROR_CODE_DESCRIPTIONS = {
+  [ApiErrorCodes.DuplicateOrder]: 'There was another identical order already submitted',
+  [ApiErrorCodes.InsufficientFee]: "The account doesn't have enough funds",
+  [ApiErrorCodes.InvalidSignature]: 'The order signature is invalid',
+  [ApiErrorCodes.MissingOrderData]: 'The order has missing information',
+  [ApiErrorCodes.InsufficientValidTo]: "The account doesn't have enough funds",
+  [ApiErrorCodes.InsufficientFunds]: "The account doesn't have enough funds",
+  [ApiErrorCodes.UnsupportedToken]: 'An unsupported token was detected',
+  [ApiErrorCodes.WrongOwner]: 'An invalid owner address was given',
+  UNHANDLED_ERROR: 'The order was not accepted by the network'
+}
+
+export default class OperatorError extends Error {
+  name = 'OperatorError'
+  type: ApiErrorCodes
+  description: ApiError['description']
```

Comment:
I find this more confusing than just adding a `string`. Imo, not worth it to reuse types in this case